pilot/pkg/model/service.go

```diff
@@ -84,6 +86,9 @@ type Service struct {
 	// or use the passthrough model (i.e. proxy will forward the traffic to the network endpoint requested
 	// by the caller)
 	Resolution Resolution
+
```

Comment:
Any reason to not add the "ID" ( == namespace + name for k8s, file path for file, etc ) ? It would really help - many times we get errors about a hostname or another, but in a multi-team environment it's very hard to know where that error comes from, since there is no good way to see where it was defined. 

Many setups restrict teams to namespaces - only admin can query across.

Reply:
@rshriram thoughts?

Reply:
What would ID contain that hostname doesn't have? hostname is already name.namespace.svc.cluster.local. ID can only be populated if we know the source from which the service is being defined. But since these thigns are being sent to k8s API servers, we don't have the original file that defined it, unless we pull it out from API server (in which case I would call it as metadata, not ID). For CRDs, its the same issue.. One serviceEntry can become multiple services. So, multiple service entries will have same ID.

Reply:
FWIW Mixer already defined the notion of service UID that is sent upstream in the control plane: https://github.com/istio/istio/blob/master/pilot/pkg/model/service.go#L371

Reply:
Yes, but the service attributes are populated only for k8s, used by RBAC.

Reply:
They are used by mixer and and all mixer adapters...